topics/compute/openstack/docs/how-to-guides/accessing-instances.md

```diff
@@ -42,6 +42,57 @@ To connect to instances running on OpenStack, you can use one of the available a
+
+        ``` bash
+        sudo useradd -m -g users <username>
+        sudo passwd <username>
+        ```
+
+        The virtual machine is almost ready for connection using the xrpd protocol. However, it should be noted that xrdp (as well as the original rdp) were not designed for encrypted communication and a machine configured in this way is not suitable for transferring sensitive information (e.g. logging into web applications used in everyday life, such as banking and the like). For some purposes, unencrypted traffic may be sufficient (`in which case please proceed to step 3`), but for greater security we recommend using SSH Tunneling described below (`in which case skip step 3 and kindly continue to step 4`).
+
+    3. Without the use of SSH Tunneling, rdp port must be open in the VM's security group. In case of using custom port number, the port is `3389`, but direct `Rule: RDP` option is awailable in the OpenStack web application. Now, it should be possible to connect with an rdp client as described in step 5.
+
+    4. Setting up SSH Tunneling takes just a few more steps:
+
+        1. Change xrdp service setting in the target VM to listen to xrpd session on localhost instead of the default external port 3389. After invoking the configuration file, change line `port=3389` to `port=tcp://.:3389`:
+
+            ``` bash
+            sudo nano /etc/xrdp/xrdp.ini
+            ```
+
+        2. Restart the xrdp service:
+
+            ``` bash
+            sudo systemctl restart xrdp
+            ```
+
+        3. Before connecting with rdp client as described below, start the SSH tunnel. Leave the terminal open, when using remote desktop. This command is run from your workstation (no longer the VM):
+
+            ``` bash
+            ssh ubuntu@[IP] -L 3399:127.0.0.1:3389
+            ```
+
+    5. If the guide above was followed, it should be possible to connect to the Ubuntu VM and use its GUI, with the use of an rdp client. RDP client is any desktop application, that understands rdp traffic and opens a window containing interactive desktop of remote host. In Windows environment, `Microsoft Remote Desktop` should already be installed inside the operating system. For MacOS `Microsoft Remote Desktop` can be downloaded from Mac App Store. Finally, if connecting from a linux workstation, there are many clients to choose from. One of the wide spread ones (this one is also tested to work) is called `Remmina`. Since the clients differ, you might benefit from documentation targeting given software, but general notes are:
+
+        - If connecting without SSH, you do not need to specify the port number and the remote machine is referred to simply as `[IP]`. With the actual floating (public) IP address.
+
+        - With the use of SSH Tunnel, connection is made to localhost with the port number chosen when opening the tunnel. Given the tunnel configuration in this documentation, the remote machine should be referred to as `localhost:3399`.
```

Comment:
Could you possibly change the lines 66-94 following way please?
 * let's show xrdp with and w/o security
   * so replace ssh tunneling with xrdp TLS encryption, see https://karthidkk123.medium.com/how-to-secure-xrdp-on-ubuntu-centos-with-ssl-tls-encryption-9258162b24a3 and https://github.com/neutrinolabs/xrdp/issues/702
   * in both cases traffic goes through rdp port 3389

People interested in remote desktop would have trouble creating ssh tunnel IMHO.